File: Common/Network/Messages/LaunchConfiguration/LaunchConfigurationRequest.cs
namespace Common.Network.Messages.LaunchConfiguration;

public class LaunchConfigurationRequest: Message
{
    public override MessageHeader GetHeader()
    {
        return new MessageHeader(MessageRegistry.GetKeyForMessageType(typeof(LaunchConfigurationRequest)), 0);
    }

    public override void ApplyData(Stream stream)
    {
        throw new NotSupportedException();
    }

    protected override Stream GetData() => new MemoryStream();
}

File: IntegrationMod/src/main/java/com/example/examplemod/LaunchConfigurationRequest.java
package com.example.examplemod;

import org.apache.http.MethodNotSupportedException;

import java.io.InputStream;

public final class LaunchConfigurationRequest extends Message {
    @Override
    public MessageHeader getMessageHeader() throws Exception {
        return new MessageHeader(MessageRegistry.getKeyForMessageType(LaunchConfigurationRequest.class), 0);
    }

    @Override
    public void applyData(InputStream stream) throws MethodNotSupportedException {
        throw new MethodNotSupportedException("That request don't support this method");
    }

    @Override
    public byte[] getData() {
        return new byte[0];
    }
}
